src/transformers/models/altclip/modeling_altclip.py

```diff
@@ -1613,7 +1613,7 @@ def forward(
         pixel_values: Optional[torch.FloatTensor] = None,
         attention_mask: Optional[torch.Tensor] = None,
         position_ids: Optional[torch.LongTensor] = None,
-        token_type_ids=None,
+        token_type_ids: Optional[torch.Tensor] = None,
```

Comment:
From my understanding, this should be a `torch.LongTensor` as type_ids are usually ints, however, if this contains a `nan` the whole array becomes a `torch.DoubleTensor` in which case the generic `torch.Tensor` will cover it. 

I kept the `torch.Tensor` for consistency with [AltCLIPTextModel](https://github.com/huggingface/transformers/blob/40e8c36170281464ace640e1984b53fb0d268c33/src/transformers/models/altclip/modeling_altclip.py#L1389)

Let me know if you want me to change it :slightly_smiling_face:

Reply:
`torch.Tensor` is fine here - I'd generally avoid passing `nan` in the `token_type_ids` inputs, though!